### ISSUE TITLE
`v-for` not rendering the items correctly

### ISSUE THREAD

**Issue by jaiko86** [vuejs/vue#10376]
### Version
2.6.10

### Reproduction link
[https://github.com/jaiko86/subtasks/pull/1](https://github.com/jaiko86/subtasks/pull/1)

### Steps to reproduce
1. Clone the project, and fetch the branch `workspace`.
2. Start the project.
3. When the app launches, click the left-most tab on the top that reads "workspace name". 
4. An empty task should be visible now. 
5. Click the empty task so the input is focused, then press enter.

### What is expected?
Another empty task should appear at the bottom

### What is actually happening?
No change happens in the view. 

---
Basically, the view is supposed to render an empty task for each item in the computed property `taskIds` in the `<App>`. When the steps to reproduce has been followed, the Vue devtool shows that `taskIds: ["1", "2"]` is in the computed property in the `<App>` component. The empty task that is currently exists on the screen corresponds to ID "1".

Go to the console and type `printTree()`. It will print the following:
```
0
  1

  2
```
It means the root ("0") has 2 children, "1" and "2".

**Comment by jaiko86** [comment]
The store is a mess, BTW, and I apologize for it.

**Comment by posva** [comment]
Please, create a minimal boiled down repro when reporting a bug
